File: backend/src/main/java/com/example/stamp_app/entity/MeasuredDataMaster.java
package com.example.stamp_app.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class MeasuredDataMaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String dayOfYear;

    @Column
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @Column
    private LocalDateTime deletedAt;

    @ManyToOne
    private MicroController microController;

    @JsonManagedReference
    @OneToMany(mappedBy = "measuredDataMaster")
    private List<Sdi12Data> sdi12Data;

    @JsonManagedReference
    @OneToMany(mappedBy = "measuredDataMaster")
    private List<EnvironmentalData> environmentalData;

    @Column
    private String voltage;

}

File: backend/src/main/java/com/example/stamp_app/entity/MeasuredDataMaster.kt
package com.example.stamp_app.entity

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
data class MeasuredDataMaster(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	val id: Long? = null,

	@Column
	val dayOfYear: String,

	@Column
	val createdAt: LocalDateTime,

	@Column
	val updatedAt: LocalDateTime,

	@Column
	val deletedAt: LocalDateTime? = null,

	@ManyToOne
	val microController: MicroController,

	@JsonManagedReference
	@OneToMany(mappedBy = "measuredDataMaster")
	val sdi12Data: List<Sdi12Data>? = null,

	@JsonManagedReference
	@OneToMany(mappedBy = "measuredDataMaster")
	val environmentalData: List<EnvironmentalData>? = null,

	@Column
	val voltage: String? = null,
)
